src/NuGet.Core/NuGet.Build.Tasks/NuGet.RestoreEx.targets

```diff
@@ -31,6 +31,7 @@ Copyright (c) .NET Foundation. All rights reserved.
         RestorePackagesConfig="$(RestorePackagesConfig)"
         SolutionPath="$(SolutionPath)"
         ProcessFileName="$(NuGetConsoleProcessFileName)"
+        MSBuildStartupDirectory="$(MSBuildStartupDirectory)"
```

Comment:
Does this mean that when our task is executing, `Directory.CurrentDirectory()` does not return the directory where the user ran msbuild?

Converting relative to absolute paths as soon as possible and relying on absolute paths throughout our codebase seems like a good idea. But it just seems weird that we can't use the runtime's built-in methods.

Reply:
It does not, because the working directory is going to the NuGet assembly that started this new process. 

Relying on absolute codepaths is nice, but this allows us to keep the code for settings resolution as *close* as possible.
Furthermore the current NuGet process is already a `forwarder` without much of it's own restore logic.